src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetailTabs/CustomLogsDropdown/CustomLogsDropdown.tsx

```diff
@@ -0,0 +1,288 @@
+            const today = moment().format('YYYY-MM-DD')
+            const dateTimeToCompare = moment(`${today}T${option.value}`)
+            if (dateTimeToCompare.isBefore(moment())) {
+                nearestTimeOption = option
+            }
+        })
+        setUnitlTime(nearestTimeOption)
+        return nearestTimeOption
+    }
+    const handleFocusChange = ({ focused }) => {
+        setFocused(focused)
+    }
+    useEffect(() => {
+        setInputError('')
+    }, [customLogsOption])
+    useEffect(() => {
+        const nearestOption = getNearestTimeOptionBeforeNow()
+        const index = ALLOW_UNTIL_TIME_OPTIONS.findIndex((option) => option === nearestOption)
+        const newOptions = excludeFutureTimingsOptions(ALLOW_UNTIL_TIME_OPTIONS, index)
+        setUntilTimeOptions(newOptions)
+    }, [])
+
+    const handleDatesChange = (selected) => {
+        setDate(selected)
+        setCustomLogsOption({ ...customLogsOption, value: getTimeStamp(selected, showUntilTime.value).toString() })
+        if (selected.isSame(moment(), 'day')) {
+            const nearestOption = getNearestTimeOptionBeforeNow()
+            const index = ALLOW_UNTIL_TIME_OPTIONS.findIndex((option) => option === nearestOption)
+            const newOptions = excludeFutureTimingsOptions(ALLOW_UNTIL_TIME_OPTIONS, index)
+            setUntilTimeOptions(newOptions)
+        } else {
+            setUntilTimeOptions(ALLOW_UNTIL_TIME_OPTIONS)
+        }
+    }
+    const handleTimeUntilChange = (selected) => {
+        setUnitlTime(selected)
+        setCustomLogsOption({ ...customLogsOption, value: getTimeStamp(date, selected.value).toString() })
+    }
+
+    const checkRequiredError = (e) => {
+        if (e.target.value === '') {
+            setInputError('This field is required')
+        } else if (inputError) {
+            setInputError('')
+        }
+    }
+
+    const offset = moment(new Date()).format('Z')
+    const timeZone = `${Intl?.DateTimeFormat()?.resolvedOptions()?.timeZone ?? ''} (GMT ${offset})`
+
+    switch (customLogsOption.option) {
+        case 'duration':
+        case 'lines':
+            return (
+                <div className="flexbox-col">
+                    <div className="dc__required-field mb-6 fs-13 fcn-7">
+                        {customLogsOption.option === 'duration' ? 'View logs for last' : 'Set number of lines'}
+                    </div>
+                    <div className="flex dc__align-start">
+                        <div className="w-180">
+                            <CustomInput
+                                error={inputError}
+                                name="duration-lines"
+                                disabled={!!inputError}
+                                handleOnBlur={checkRequiredError}
+                                value={input}
+                                rootClassName="input-focus-none"
+                                onChange={(e) => {
```

Comment:
Please extract this to separate function

Reply:
done